kubectl: add v1.30.0 -> v1.32.3

RobertMaaskant opened the pull request:
Added all versions currently supported upstream: last three minor versions. Check [https://kubernetes.io/releases/](https://kubernetes.io/releases/) for details.

Only did a some basic checks for the added versions, but don't see how the newer versions would be broken.

Checks done:
- compiled correctly
- run a few basic commands
- check correct Go version dependency (based on `go.mod` upstream)

Would appreciate feedback on my usage of `deprecated=True`. Especially for the existing 1.27.x versions. Still kind of new to Spack, so this might not be the desired way to do things.

RobertMaaskant committed:
kubectl: add all versions currently supported upstream

tldahlgren reviewed (commented):
FYI. Confirmed all affected version sha256s.

RobertMaaskant committed:
kubectl: build same way as kubernetes

RobertMaaskant commented:
Added version 1.27.2 to bring in sync with package `kubernetes`.

Also changed the build to the same method as in package `kubernetes`. This fixes a bug where `kubectl version` did not display the correct client version.

Before:
```
$ kubectl version
Client Version: v0.0.0-master+67a30c0adcf52bd3f56ff0893ce19966be12991f
Kustomize Version: v5.5.0
The connection to the server localhost:8080 was refused - did you specify the right host or port?
```

After:
```
$ kubectl version
Client Version: v1.32.2
Kustomize Version: v5.5.0
The connection to the server localhost:8080 was refused - did you specify the right host or port?
```

alecbcs commented:
`kubectl` was intentionally moved to being a `GoPackage` to fix issues with the go module cache, I haven't gotten to updating `kubernetes` yet but that one should also be updated to be a go package soon. Please revert the package and I'll review the added versions.

RobertMaaskant committed:
kubectl: revert back to GoPackage

RobertMaaskant committed:
kubectl: fix version command

RobertMaaskant commented:
Reverted back to a `GoPackage` as requested and added build args to correctly set the version.

RobertMaaskant commented:
@alecbcs, I made the requested changes. Are we ready to merge now?

And if you want, I could take a look at changing the kubernetes Package to inherit from the GoPackage.

RobertMaaskant committed:
kubectl: add v1.30.11, v1.31.7, v1.32.3

RobertMaaskant commented:
Add new versions:
- 1.30.11
- 1.31.7
- 1.32.3

Release notes:
- https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.30.md#v13011
- https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.31.md#v1317
- https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.32.md#v1323

tldahlgren commented:
@alecbcs Any objections to this being merged?

alecbcs commented:
Yes. Typically we don't add deprecated versions that aren't already in the package. 

@RobertMaaskant the changes to the versioning flags look good! 👍 We typically don't add versions to the package as "deprecated" unless you really need them to block people from installing CVEs. If you can pull out the version checksums you added newly in this package as deprecated that'd be great!

RobertMaaskant committed:
kubectl: remove new deprecated versions

RobertMaaskant committed:
kubectl: refactor build deps

RobertMaaskant commented:
@alecbcs thank you for another round of review!

Removed the new deprecated versions.

alecbcs reviewed (approved):
Looks good to me. Thanks @RobertMaaskant for the quick turnaround.
